integration-tests/src/tests/transaction_payment.rs

```diff
@@ -1,4 +1,181 @@
+use pallet_transaction_payment::FeeDetails;
+use parity_scale_codec::Encode;
+use polimec_common::{ProvideAssetPrice, PLMC_DECIMALS, PLMC_FOREIGN_ID, USD_DECIMALS};
+use polimec_runtime::Header;
+use sp_arithmetic::{FixedPointNumber, FixedU128};
+use sp_core::H256;
+use sp_runtime::{
+	traits::{Dispatchable, Header as HeaderT, SignedExtension},
+	DigestItem,
+};
+use xcm_emulator::TestExt;
+generate_accounts!(ALICE, AUTHOR);
+use frame_support::traits::fungible::Inspect;
+
+// Setup code inspired by pallet-authorship tests
+fn seal_header(mut header: Header, aura_index: u64) -> Header {
+	{
+		let digest = header.digest_mut();
+		digest.logs.push(DigestItem::PreRuntime(sp_consensus_aura::AURA_ENGINE_ID, aura_index.encode()));
+		digest.logs.push(DigestItem::Seal(sp_consensus_aura::AURA_ENGINE_ID, aura_index.encode()));
+	}
+
+	header
+}
+
+fn create_header(number: u32, parent_hash: H256, state_root: H256) -> Header {
+	Header::new(number, Default::default(), state_root, parent_hash, Default::default())
+}
+// Make sure to run this inside externalities environment. Can only be called one time per test.
+fn set_author(aura_index: u64) {
+	let mut header = seal_header(create_header(1, Default::default(), [1; 32].into()), aura_index);
+
+	header.digest_mut().pop(); // pop the seal off.
+	PolimecSystem::reset_events();
+	PolimecSystem::initialize(&1, &Default::default(), header.digest());
+}
+
 #[test]
 fn fee_paid_with_foreign_assets() {
-	todo!();
+	polimec::set_prices(
+		PricesBuilder::new().usdt(FixedU128::from_float(1.0f64)).plmc(FixedU128::from_float(0.5f64)).build(),
+	);
+
+	PolimecNet::execute_with(|| {
+		let alice: PolimecAccountId = ALICE.into();
+
+		let (block_author, _) = &constants::collators::invulnerables()[1];
+		// Block author's aura index is 1
+		set_author(1u64);
+		assert_eq!(polimec_runtime::Authorship::author(), Some(block_author.clone()));
+
+		let usdt_id = AcceptedFundingAsset::USDT.id();
+		let usdt_decimals = <PolimecForeignAssets as fungibles::metadata::Inspect<PolimecAccountId>>::decimals(usdt_id);
+		let usdt_unit = 10u128.pow(usdt_decimals as u32);
+		let plmc_decimals = PLMC_DECIMALS;
+		let plmc_unit = 10u128.pow(plmc_decimals as u32);
+
+		PolimecBalances::set_balance(&alice, 0u128);
+		PolimecForeignAssets::set_balance(usdt_id, &alice, 100 * usdt_unit);
+		// Fees are usually very small, so we need to give the treasury an ED.
+		PolimecForeignAssets::set_balance(
+			usdt_id,
+			&polimec_runtime::BlockchainOperationTreasury::get(),
+			100 * usdt_unit,
+		);
+		// Block author doesn't need to have any balance, as the tip is bigger than ED.
+
+		let paid_call = PolimecCall::System(frame_system::Call::remark { remark: vec![69, 69] });
+
+		type TxPaymentExtension = pallet_asset_tx_payment::ChargeAssetTxPayment<PolimecRuntime>;
+		let signed_extension =
+			pallet_asset_tx_payment::ChargeAssetTxPayment::<PolimecRuntime>::from(10 * plmc_unit, Some(usdt_id));
```

Comment:
Can you please clarify with a code comment what's happening on this line? Is it correct to say that "a user wants to leave as a tip the equivalent in USDT of 10 PLMC"?